### PR TITLE
Feature/ingredient repo class and tests

### LINKED ISSUE #30: Build IngredientRepo Class
Test:
- [x] All class properties
- [x] All methods and updates to properties
- [x] Any helper methods you build out
- [x] Happy/sad path tests
- [x] Create own data to test

Should have properties:
- [x] constructor(IngredientData) - parameter to take in all ingredient `data`
- [x] ingredients (list of ingredients as Ingredient objects from ingredient data)

Methods:
- [x] `returnIngredientById()` - return a filtered list of ingredients based on ids, used to match ingredient data with recipe ingredient properties based on their ids, connecting data to recipes

### PR DESCRIPTION
**Pull request by elizhahn**
## Is this a fix or a feature?
- feature

## What is the change?
- This adds an IngredientRepo class and testing suite
## What does it fix?
- N/A
## Where should the reviewer start?
- ingredientRepo-test.js
- ingredientRepo.js
## How should this be tested?
- run npm test in root directory, all tests should be passing

### CODE REVIEW

File: test/ingredientRepo-test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+const chai = require('chai');
+const expect = chai.expect;
+const Ingredient = require('../src/ingredient.js');
+const IngredientRepo = require('../src/ingredientRepo.js');
+
+describe('Ingredient Repo', () => {
+  let ingredients;
+  beforeEach(() => {
+    ingredientData = [
```
Comment (alexthompson207) [inline test/ingredientRepo-test.js:9]: declare  ingredientData on line 7

File: test/ingredientRepo-test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const chai = require('chai');
+const expect = chai.expect;
+const Ingredient = require('../src/Ingredient');
+const IngredientRepo = require('../src/IngredientRepo');
+
+describe('Ingredient Repo', () => {
+  let ingredientList;
+  let defaultIngredientList;
```
Comment (alexthompson207) [inline test/ingredientRepo-test.js]: refactor line 7 to say `let ingredientList, defaultIngredientList, ingredientData;`

File: test/ingredientRepo-test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  beforeEach(() => {
+    const ingredientData = [
```
Comment (alexthompson207) [inline test/ingredientRepo-test.js]: Remove the const, since we are declaring it above

File: test/ingredientRepo-test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  it('should have no Ingredients by default', () => {
+    expect(defaultIngredientList.ingredients).to.deep.equal([]);
+  });
+
+  it('should hold a list of Ingredients', () => {
+    expect(ingredientList.ingredients[0]).to.be.an.instanceof(Ingredient);
+  });
+
+  it('should return an ingredient id', () => {
```
Comment (alexthompson207) [inline test/ingredientRepo-test.js:56]: Maybe make a sad path, if ids don't match--- return undefined?

**Comment by alexthompson207** [comment]
Looks great! Good work!

File: test/ingredientRepo-test.js (hunk 4 above)
Comment (elizhahn) [reply test/ingredientRepo-test.js:56]: Good point I will def add this

**Pull request merged**